src/Hl7.Fhir.Support.Tests/FhirPath/FhirPathTests.cs

```diff
@@ -81,5 +81,81 @@ public static IEnumerable<object[]> GetTypedElements()
             yield return new object[] { poco.ToTypedElement(ModelInspector.Base), "poco to TypedElement" };
 
         }
+
+        public static IEnumerable<object[]> LowBoundaryTestCases() =>
+            new (string expression, bool expected)[]
+                {
+                    ("1.587.lowBoundary(8) = 1.58650000", true),
+                    ("1.587.lowBoundary()  = 1.5865", true),
+                    ("0.0.lowBoundary()    = -0.05", true),
+                    ("1.1.lowBoundary(8)   = 1.05000000", true),
+                    ("1.25.lowBoundary(8)  = 1.24500000", true),
+                    ("1.0.lowBoundary(8)   = 0.95000000", true),
+                    ("2400.lowBoundary(8)  = 2399.50000", true),
+                    ("2.25.lowBoundary(8)  = 2.24500000", true),
+                    ("2.2500.lowBoundary(8)= 2.24995000", true),
+                    ("12.lowBoundary() = 12", true),
+                    ("'string'.lowBoundary() = 'string'", false),
+                    ("1.587 'cm'.lowBoundary(8) = 1.58650000 'cm'", true),
+                    ("@2014.lowBoundary(6) = @2014-01", true),
+                    ("@2014.lowBoundary() = @2014-01-01T00:00:00.000", true),
+                    ("@2014.lowBoundary(8) = @2014-01-01", true),
+                    ("@2014-01-01T08.lowBoundary(17) = @2014-01-01T08:00:00.000", true),
+                    ("@T10:30.lowBoundary(9) = @T10:30:00.0001", true),
```

Comment:
Why isn't this just ending with `.0000`?  Just proves that I don't really understand what this is doing I guess ;-)